proposals/0191-enable-transaction-loading-failure-fees.md

```diff
@@ -67,6 +67,30 @@ constraints may be included in a block, so long as it is otherwise valid.
 The transaction must pay transaction fees, and if present, the nonce must be
 advanced.
 
+Constraints must be checked before committing transactions and voting.
+It is reccomended that the constraints are checked before transaction
+execution, in order to avoid unnecessary computation.
```

Comment:
How about

> Constraints SHOULD be checked for each transaction before execution to avoid unnecessary computation. If a constraint violating transaction is executed, the constraints MUST be checked BEFORE committing transaction changes.

I don't exactly get why you said "and voting".. is specifying before committing sufficient?